ui/store/actions.ts

```diff
@@ -4771,6 +4772,38 @@ export function setSnapsInstallPrivacyWarningShownStatus(shown: boolean) {
     );
   };
 }
+
+/**
+ * Update the state of a given Snap interface.
+ *
+ * @param id - The Snap interface ID.
+ * @param state - The interface state.
+ * @returns Promise Resolved on successfully submitted background request.
+ */
+export function updateInterfaceState(
+  id: string,
+  state: InterfaceState,
+): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
+  return (async (dispatch: MetaMaskReduxDispatch) => {
+    await submitRequestToBackground<void>('updateInterfaceState', [id, state]);
+    forceUpdateMetamaskState(dispatch);
+  }) as any;
```

Comment:
Why do we need to cast here?

Reply:
Because the type is really weird here, I'm not sure why but all the others has the same trick